mmv1/third_party/terraform/functions/element_from_id.go

```diff
@@ -1,36 +1,29 @@
 package functions
 
 import (
+	"context"
 	"fmt"
 	"regexp"
 
 	"github.com/hashicorp/terraform-plugin-framework/function"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
 )
 
-const noMatchesErrorSummary string = "No matches present in the input string"
-const ambiguousMatchesWarningSummary string = "Ambiguous input string could contain more than one match"
-
-// ValidateElementFromIdArguments is reusable validation logic used in provider-defined functions that use the getElementFromId function
-func ValidateElementFromIdArguments(input string, regex *regexp.Regexp, pattern string, resp *function.RunResponse) {
+// ValidateElementFromIdArguments is reusable validation logic used in provider-defined functions that use the GetElementFromId function
+func ValidateElementFromIdArguments(ctx context.Context, input string, regex *regexp.Regexp, pattern string) *function.FuncError {
 	submatches := regex.FindAllStringSubmatchIndex(input, -1)
 
 	// Zero matches means unusable input; error returned
 	if len(submatches) == 0 {
-		resp.Diagnostics.AddArgumentError(
-			0,
-			noMatchesErrorSummary,
-			fmt.Sprintf("The input string \"%s\" doesn't contain the expected pattern \"%s\".", input, pattern),
-		)
+		return function.NewArgumentFuncError(0, fmt.Sprintf("The input string \"%s\" doesn't contain the expected pattern \"%s\".", input, pattern))
 	}
 
-	// >1 matches means input usable but not ideal; issue warning
+	// >1 matches means input usable but not ideal; debug log
 	if len(submatches) > 1 {
-		resp.Diagnostics.AddArgumentWarning(
-			0,
-			ambiguousMatchesWarningSummary,
-			fmt.Sprintf("The input string \"%s\" contains more than one match for the pattern \"%s\". Terraform will use the first found match.", input, pattern),
-		)
+		tflog.Debug(ctx, fmt.Sprintf("The input string \"%s\" contains more than one match for the pattern \"%s\" expected by the provider function. Terraform will use the first found match.", input, pattern))
```

Comment:
If there was a way to access the function name here it'd be good to include that in the log

Reply:
Implemented in 3dd88b4